soroban-sdk/src/storage.rs

```diff
@@ -139,6 +139,15 @@ impl Storage {
         }
     }
 
+    /// Returns the maximum number of ledgers that an entry can live.
+    pub fn max_expiration(&self) -> u32 {
+        let seq = self.env.ledger().sequence();
+        let max: u32 = internal::Env::get_max_expiration_ledger(&self.env)
+            .unwrap_infallible()
+            .into();
+        max - seq
```

Comment:
Need to add `+ 1` due to how math works for this (max entry lifetime includes the current ledger)

Reply:
Fixed in d42a312.